plugins/inputs/sqlserver/azuresqlpoolqueries.go

```diff
@@ -0,0 +1,513 @@
+	SET @ErrorMessage = 'Telegraf - Database:' + DB_NAME() +' is not attached to an elastic pool. Check the database_type parameter in the telegraf configuration.';
+	RAISERROR (@ErrorMessage,11,1)
+	RETURN
+END
+
+SELECT TOP(1)
+   'sqlserver_pool_resource_stats' AS [measurement]
+  ,REPLACE(@@SERVERNAME,'\',':') AS [sql_instance]
+  ,(SELECT [elastic_pool_name] FROM sys.database_service_objectives) AS [elastic_pool_name]
+  ,[snapshot_time]
+  ,cast([cap_vcores_used_percent] as float) AS [avg_cpu_percent]
+  ,cast([avg_data_io_percent] as float) AS [avg_data_io_percent]
+  ,cast([avg_log_write_percent] as float) AS [avg_log_write_percent]
+  ,cast([avg_storage_percent] as float) AS [avg_storage_percent]
+  ,cast([max_worker_percent] as float) AS [max_worker_percent]
+  ,cast([max_session_percent] as float) AS [max_session_percent]
+  ,[max_data_space_kb]/1024 AS [storage_limit_mb]
+  ,cast([avg_instance_cpu_percent] as float) AS [avg_instance_cpu_percent]
+  ,cast([avg_allocated_storage_percent] as float) AS [avg_allocated_storage_percent]
+FROM 
+  sys.dm_resource_governor_resource_pools_history_ex WITH (NOLOCK)
+WHERE 
+  [name] = 'SloSharedPool1'
+ORDER BY
+  [snapshot_time] DESC;
+`
+
+const sqlAzurePoolResourceGovernance = `
+DECLARE @ErrorMessage AS nvarchar(500)
+IF SERVERPROPERTY('EngineEdition') <> 5 BEGIN /*not Azure SQL DB*/
+	SET @ErrorMessage = 'Telegraf - Connection string Server:'+ @@SERVERNAME + ',Database:' + DB_NAME() +' is not an Azure SQL DB. Check the database_type parameter in the telegraf configuration.';
+	RAISERROR (@ErrorMessage,11,1)
+	RETURN
+END
+IF (SELECT [elastic_pool_name] FROM sys.database_service_objectives) IS NULL BEGIN
+	SET @ErrorMessage = 'Telegraf - Database:' + DB_NAME() +' is not attached to an elastic pool. Check the database_type parameter in the telegraf configuration.';
+	RAISERROR (@ErrorMessage,11,1)
+	RETURN
+END
+
+SELECT
+	 'sqlserver_pool_resource_governance' AS [measurement]
+	,REPLACE(@@SERVERNAME,'\',':') AS [sql_instance]
+	,(SELECT [elastic_pool_name] FROM sys.database_service_objectives) AS [elastic_pool_name]
+	,[database_name]
+	,[slo_name]
+	,[dtu_limit]
+	,[cpu_limit]
+	,[max_cpu]
+	,[cap_cpu]
+	,[max_db_memory]
+	,[max_db_max_size_in_mb]
+	,[db_file_growth_in_mb]
+	,[log_size_in_mb]
+	,[instance_cap_cpu]
+	,[instance_max_log_rate]
+	,[instance_max_worker_threads]
+	,[checkpoint_rate_mbps]
+	,[checkpoint_rate_io]
+	,[primary_group_max_workers]
+	,[primary_min_log_rate]
+	,[primary_max_log_rate]
+	,[primary_group_min_io]
+	,[primary_group_max_io]
+	,[primary_group_min_cpu]
+	,[primary_group_max_cpu]
+	,[primary_pool_max_workers]
+	,[pool_max_io]
+	,[volume_local_iops]
+	,[volume_managed_xstore_iops]
+	,[volume_external_xstore_iops]
+	,[volume_type_local_iops]
+	,[volume_type_managed_xstore_iops]
+	,[volume_type_external_xstore_iops]
+	,[volume_pfs_iops]
+	,[volume_type_pfs_iops]
+FROM 
+	sys.dm_user_db_resource_governance WITH (NOLOCK);
+`
```

Comment:
Fixed.

Reply:
I think accross all the measures same DMV = same measure, this is a bit different ( aka sqlserver_pool_resource_governance)  in that it has the same DMV ( sys.dm_user_db_resource_governance) but has a different measure name. For MI I get that the measure name is different ( but then so is the DMV).

Reply:
For all measures, I used similar names used by existing queries, except for resource stats and governance as they were not aligned for MI and Single database. I don't know what is the best practice here, and I can change these names if you prefer. However, using sqlserver_azure_db_resource_stats / sqlserver_db_resource_governance can be also confusing as we don't target a database but a pool.

AzureSql -> sqlserver_azure_db_resource_stats / sqlserver_db_resource_governance
ManagedInstance -> sqlserver_azure_db_resource_stats / sqlserver_instance_resource_governance
For Pool, I choose -> sqlserver_pool_resource_stats/ sqlserver_pool_resource_governance

Reply:
Yes. I get your point for naming convention. To be honest, I wanted to prefix every measure names with azuresql_pool_* to differentiate them.  But to be aligned with existing queries, I used existing names except for resource stats and resource governance.  

For resource stats, what we tried to achieve is to reproduce the view from **sys.elastic_pool_resource_stats**. So I updated existing name from sqlserver_azure_db_resource_stats to  sqlserver_pool_resource_stats. (Note: Weird to have **azure** in existing name, as it's not the case for other queries).

For governance, we retrieve the governance info of the pool, not a specific database. Elastic pool name is returned instead database name. Even this is the same DMV, I preferred to use **pool** in the name instead **db**.   

For IO, as we have results for all databases in the pool, I didn't change the name. 

sys.dm_resource_governor_resource_pools_history_ex -> **sqlserver_pool_resource_stats**
sys.dm_user_db_resource_governance -> **sqlserver_pool_resource_governance**
sys.dm_io_virtual_file_stats -> sqlserver_database_io
sys.dm_os_wait_stats -> sqlserver_waitstats
sys.dm_os_memory_clerks -> sqlserver_memory_clerks
sys.dm_os_performance_counters -> sqlserver_performance
sys.dm_os_schedulers -> sqlserver_schedulers

Anyway, I am totally open to change the names. What are your recommendations ?

Reply:
No I don't think you should change any names other than new DMV's collected as it will break visualizations folks have as they depend on the measures.

Reply:
Ok, we don't change current measure names.